### ISSUE TITLE
No blockchain is provided to stranger Ursulas when learning

### ISSUE THREAD

**Issue by cygnusv** [nucypher/nucypher#1202]
In decentralized mode, no blockchain is given to stranger Ursulas during learning, which ends up in an error when trying to instantiate the Ursula. 

```File "/home/leeloo/nucypher/lib/python3.6/site-packages/nucypher/characters/lawful.py", line 1054, in from_rest_url
    *args, **kwargs)
  File "/home/leeloo/nucypher/lib/python3.6/site-packages/nucypher/characters/lawful.py", line 1229, in from_bytes
    ursula = cls.from_public_keys(federated_only=federated_only, **node_info)
  File "/home/leeloo/nucypher/lib/python3.6/site-packages/nucypher/characters/base.py", line 311, in from_public_keys
    return cls(is_me=False, federated_only=federated_only, crypto_power=crypto_power, *args, **kwargs)
  File "/home/leeloo/nucypher/lib/python3.6/site-packages/nucypher/characters/lawful.py", line 878, in __init__
    **character_kwargs)
  File "/home/leeloo/nucypher/lib/python3.6/site-packages/nucypher/characters/base.py", line 134, in __init__
    self.staking_agent = StakingEscrowAgent(blockchain=blockchain)
  File "/home/leeloo/nucypher/lib/python3.6/site-packages/nucypher/blockchain/eth/agents.py", line 42, in __call__
    cls.__agents[cls] = super().__call__(*args, **kwargs)
  File "/home/leeloo/nucypher/lib/python3.6/site-packages/nucypher/blockchain/eth/agents.py", line 85, in __init__
    raise ValueError("Blockchain is required to connect a new agent.")
```

**Comment by michwill** [comment]
Excellent catch!

**Comment by cygnusv** [comment]
A variant of this problem is affecting `__read_metadata` in `NodeStorage`, and currently both `nucypher alice` and `nucypher bob` are unusable.

**Comment by KPrasch** [comment]
closed by #1159
